# sample of 10 random edits for each bot


### subsetting

In [1]:
%reload_ext autoreload
%autoreload 2
%matplotlib inline

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

from datetime import datetime

In [3]:
df = pd.read_csv('revs_scored_jan.tsv', sep='\t', header=0)

days = df.rev_timestamp.map(lambda ts: datetime.utcfromtimestamp(ts).day)
df['day'] = days

In [4]:
df_bots = df[df.day<22]

df_cyberbot_i = df_bots[df_bots.user_text == "Cyberbot I"]
df_avicbot = df_bots[df_bots.user_text == "AvicBot"]
df_anomiebot = df_bots[df_bots.user_text == "AnomieBOT"]
df_ronbot = df_bots[df_bots.user_text == "RonBot"]

## AvicBot

In [5]:
df_avicbot.sample(n=10, random_state=10)[["rev_id", "page_id", "is_reverted", "page_namespace", "damaging_pred"]].reset_index()

index     rev_id   page_id  is_reverted  page_namespace  damaging_pred
0   902342  877167177  47909226         True               2       0.028723
1  1715559  878011638  47909226        False               2       0.007404
2   780518  877040847  47909226        False               2       0.006066
3    22051  876250007  47909226        False               2       0.012336
4   673302  876929423  52628498        False               2       0.015032
5  1717842  878014014  52628498        False               2       0.007148
6   839212  877101918  47909226        False               2       0.009779
7  1554330  877843975  47909226        False               2       0.006204
8  2715351  879055197  47909226        False               2       0.004804
9   822989  877085068  47909226        False               2       0.006802

0. 877167177 - Seems like Avicbot is just reoving users from the "Candidates for speedy deletion" Category, a minor edit, removing users, switching positions
1. 878011638 - doing the same thing here, minor edit, adding users and switching positions
2. 877040847 - same thing here, adding and removing users; like the previous list, it seems as though most of the candidates on this list no longer exist (as in the page no longer exist)
3. 876250007 - 5 mins from its previous edit, it added a user to this category again, only a few of these pages exist today
4. 876929423 - 5 mins from its previous edit, removes a user from the "All articles proposed for deletion" category, like the lists above only a few of these pages exist today, seems as AvicBot makes edits every 5 mins?
5. 878014014 - these edits aren't 5 mins apart, but it removed some articles and then added a bunch to the same category, also called a minor edit
6. 877101918 - 5 mins from its previous edit, avicbot switches and added some users, to the "Candidates for speedy deletion" category again, so it checks this category every 5 mins 
7. 877843975 - 5 mins from its prevoius edits again, switched and adds some users to the category above, still a minor edit, and like the lists above, most of these pages don't exist
8. 879055197 - 5 mins later, removes and then adds some users to the same category, still a minor edit, but unlike the last list, most of these pages exist (some of the pages are categories, so they link to a page that tells us pages under that category!)
9. 877085068 - 5 mins later, switched and adds some user to the same category, still a minor eidt, and most of these pages don't exist!

So it seems like candidates on this page were tagged with speedy deletio templates, which is put on ones page, based on them meeting the criteria for speedy deletion, https://en.wikipedia.org/wiki/Category:Candidates_for_speedy_deletion
"Speedy does not mean immediate. Although many speedy-deletion candidates are processed within a few minutes, more complicated pages may remain in the list for several days." - make sense why its checked every five minutes 
The categories above are hidden

## Cyberbot I

In [6]:
df_cyberbot_i.sample(n=10, random_state=10)[["rev_id", "page_id", "is_reverted", "page_namespace", "damaging_pred"]].reset_index()

index     rev_id   page_id  is_reverted  page_namespace  damaging_pred
0   604511  876857914    352651        False               4       0.006925
1   166853  876400051  45514171        False             109       0.008596
2    25927  876254025  39008263        False              10       0.008977
3  1579570  877870409  42414676        False             109       0.006855
4   158789  876391750  35159700        False               2       0.007126
5  2928072  879276370  21083350        False              10       0.010086
6  2785160  879127746  48564490         True              10       0.010453
7  2960799  879310786  35159700        False               2       0.013086
8   839153  877101856  16283969        False               4       0.011849
9  2012462  878321999  45160445        False              10       0.010055

0. f on RfPP page, re-ordered requests according to time stamps
1. 876400051 part of bookreport task. changing article status (to what their pages currently say?) and adds "|non-free=" to template
2. 876254025 updating someone's admin stats
3. 877870409 part of bookreport task. removed a gif from the report (can't be included in a book?)
4. 876391750 on RfX report, updated count for a vote and timestamp of latest update
5. 879276370 updating someone's admin stats
6. 879127746 updating someone's admin stats, except ***note: possible malfunction, stats go way down*** month long break since last edit. they later go back up in the next edit. as I skip through edits, they keep going up and down a bit, but never by as much as in the original edit
7. 879310786 adds RfB to RfX page and deletes line that says there are no current RfXs
8. 877101856 clears sandbox of all content, preserving header
9. 878321999 updating admin stats


## RonBot

In [11]:
df_ronbot.sample(n=13, random_state=10)[["rev_id", "page_id", "is_reverted", "page_namespace", "damaging_pred"]].reset_index()

index     rev_id   page_id  is_reverted  page_namespace  damaging_pred
0    938396  877204293  29482903         True               0       0.007589
1   1193338  877469258  25192152        False               0       0.011086
2    938364  877204258  53196630        False               0       0.008103
3   1469702  877756406  41954465        False               3       0.004873
4   2434034  878761985  36258259         True               0       0.011845
5   1009381  877278272   1194293         True               0       0.008515
6   2435827  878763867  53610237         True               0       0.012684
7    937242  877203083  58903100         True               0       0.008654
8   2436977  878765075  20776676         True               0       0.014437
9   1617506  877910061  59634955         True               6       0.005974
10  3062659  879416516  59587499        False               6       0.006930
11  3068418  879422608  59593123        False               6       0.008230
12   935504  877201314  52471236        False               0       0.018021

0. 877204293 - Even though on the players page it says "[[Category:Association football forwards]]" ronbot still adds this user to the category " Association footballers not categorized by position", so no surprise that it was reverted, however this was on january 7 0and this malfunction was reported
1. 877469258 - so January 7 when this malfunction of Ronbot was happening it added this footballer to the category above, but the next day, removed it, seems like the operator fixed it!?!
2. 877204258 - this player doesn't seem to have a position on the page, so the bot added the player to the category on August 2 2018, and then added it to the category again on January 7, definetely a malfunction, however this wasn't reverted!
3. 877756406 - seems like this user uploaded a nonfree file, so ronbot added the {{}} template on the page twice at the same time to two different images. If we scroll down the page, we can see ronbot added this template before to a bunch of other images. An automated notice was also left by another bot about a "image without license" so this doesn't seem like malfunction
4. 878761985 - on august 7 2018, ronbot added this player to the "Association footballers not categorized by position" category, even though it seems like this player is a forward. No edits in between, a year late on January 16 2019 (when i become 18 !), the bot then removed it , so I'm assuming it took a while for Ronbot to correct its mistakes, this is a revert
5. 877278272 - just a edit of a malfuncttion again, even though it says this player is a football defender (last edit was nov. 2018), ronbot added it to the the category above again, so no wonder why this page got reverted
6. 878763867 - similar to number 4, ronbot added this user to the category above on aug. 2018, even though they have a position, and then removed them from the category in jan. 16 2019, so this must be the revert!
7. 877203083 -  doesn't seem like this player has a position, but it seems like ronbot already added the user to this category sometime before oct. 2018, and then ronbot added it again on jan. 7 - no suprise that it was reverted :)
8. 878765075 - this player is a forward, even though ronbot added the user to the category sometime before sept. 2018, and then the bot removed it on jan. 16 2019, seem slike this is the revert
9. 877910061 - apparently the revision of this difference was not found, maybe because of an outdated diff link to a page or the revision has been deleted! interesting...this edit was also reverted
10. 879416516 - on jan. 7 2019, theos little bot tagged this page with a {{Non-free reduced}} template, then about a week later ronbot removes it, because it's an "orphaned" non-free file, this revision wasn't reverted, so maybe they're working together?
11. 879422608 - same thing here, theo added the same tag on jan. 8 and then ronbot removed it (task 1), also wasn't reverted...seems like colloboration to me 
12. 877201314 - another edit of ronbots malfunction; added this page to the category even though they're a goalkeeper (on jan. 7 too); it says this page wasn't reverted, but it was reverted a day later... (see https://en.wikipedia.org/w/index.php?title=Aaron_Ramsdale&diff=prev&oldid=877474074)

## AnomieBOT

In [12]:
df_anomiebot.sample(n=10, random_state=10)[["rev_id", "page_id", "is_reverted", "page_namespace", "damaging_pred"]].reset_index()

index     rev_id   page_id  is_reverted  page_namespace  damaging_pred
0  1694032  877989338  13147974        False               4       0.003473
1  3139299  879496610  33952838        False               2       0.007705
2  1257422  877535586   1237344        False               0       0.014971
3  3179797  879538929  39957309        False               0       0.018338
4   704193  876961545  59168844        False               0       0.034009
5  1791664  878090748  59513841        False               0       0.036503
6   814570  877076276  36753941        False               0       0.015101
7  1381119  877664106   9236158        False               0       0.014542
8  2019738  878329602  13184870        False               4       0.003932
9  2559142  878892635  45443583        False               0       0.028584

0. 877989338 comment: archiving closed XfD's from Wikipedia:WikiProject Deletion sorting/United Kingdom (errors?). bot adds two entries to the list of articles for deletion. edit is on page: WP:Wikiproject deletion sorting/United Kingdom/archive

1. 879496610 comment: update table (69 requests). bot changes a number from 70 to 79, page mentions WP semi-protected edit requests. edit is on page: User:AnomieBOT/SPERTable. bot also removes a few lines of the table (an edit that was completed?) and changed the 'last updated' time.

2. 877535586 comment: Dating maintenance tags: {{Definition needed}} {{Copy edit-inline}} {{Refimprove section}} {{Unreferenced section}}. bot dates 8 different maintenance tags that had no date to say Jan 2019. edit is on page: (article namespace) Nightly Business Report.

3. 879538929 comment: Dating maintenance tags: {{More sources}} {{Cn}}. bot dates 7 maintenance tags with no date. page is in article namespace

4. 876961545 comment: Dating maintenance tags: {{Copy edit}}. bot adds date to 1 template. page is in main article namespace

5. 878090748 comment: Dating maintenance tags: {{Orphan}}. bot adds date to 1 maintenance tag with no date. page is in main article npamespace.

6. 877076276 comment: Dating maintenance tags: {{Out of date}}. bot adds date to 1 maintenance tag with no date. page is in main article namespace.

7. 877664106 comment: Dating maintenance tags: {{Or}}. bot dates 1 maitenance tag with no date. page is in article namespace.

8. 878329602 comment: Archiving closed XfDs from Wikipedia:WikiProject Deletion sorting/Cricket (errors?). bot adds one article to this list. page is WP: Wikiproject deletion sorting/Cricket/archive

9. 878892635 comment: Dating maintenance tags: {{Expand section}}. bot changes maintenance tag date (which is present) to have an equals sign instead of a hyphen (hyphen probably breaks the parameter). page is in main article namespace.